src/plugins/dashboard/public/application/actions/clone_panel_action.tsx

```diff
@@ -89,61 +90,108 @@ export class ClonePanelAction implements Action<ClonePanelActionContext> {
     );
   }
 
-  private async getUniqueTitle(rawTitle: string, embeddableType: string): Promise<string> {
+  private async getCloneTitle(embeddable: IEmbeddable, rawTitle: string) {
     const clonedTag = dashboardClonePanelAction.getClonedTag();
     const cloneRegex = new RegExp(`\\(${clonedTag}\\)`, 'g');
     const cloneNumberRegex = new RegExp(`\\(${clonedTag} [0-9]+\\)`, 'g');
     const baseTitle = rawTitle.replace(cloneNumberRegex, '').replace(cloneRegex, '').trim();
 
-    const similarSavedObjects = await this.core.savedObjects.client.find<SavedObject>({
-      type: embeddableType,
-      perPage: 0,
-      fields: ['title'],
-      searchFields: ['title'],
-      search: `"${baseTitle}"`,
+    let similarTitles: string[];
+    if (isReferenceOrValueEmbeddable(embeddable)) {
+      if (rawTitle === '') return '';
```

Comment:
Nit: Could this check be moved outside of the `if (isReferenceOrValueEmbeddable(embeddable))` statement? I feel like embeddables which are not RefOrVal embeddables, _but_ also don't have a title set could potentially benefit from this early return.

Reply:
I worry that this might result in unexpected behaviour when trying to clone `search` embeddables that have had their title set to the empty string - I'll check this out and see what happens if I move it 👍

Reply:
So I checked this out, and it doesn't seem to make a difference - since `search` embeddables are *always* by value, the `rawTitle` will *always* be the library string (which cannot be empty). So the case of `rawTitle === ''` can **only** happen for RefOrVal embeddables - therefore, I don't think there is much of a point moving this logic out 🤔

Reply:
My thought here was that embeddable types which are `by value` and **only** `by value`, do not implement `ReferenceOrValueEmbeddable`. Check out `ContactCardEmbeddable` for an example of that. Its input does not contain a reference to a saved object so it's technically by value, but it also doesn't implement `ReferenceOrValueEmbeddable`. I know there aren't any embeddables like this in production, but it is hypothetically possible. 

In the current code, if there was a `ContactCardEmbeddable` on the dashboard, and it was cloned, the clone action would search the library for similar titles when it should actually be searching within the dashboard. So maybe what needs to change is: 

```typescript
if (isReferenceOrValueEmbeddable(embeddable) || !panelToClone.explicitInput.savedObjectId) {
...
}
```

Reply:
Ahhh, great point! These changes were added in 999ee47d00fa1d6cdf55cd03e14b5dc8d26a1937 :)